docs/root/intro/arch_overview/operations/overload_manager.rst

```diff
@@ -12,3 +12,43 @@ upstream services.
 The overload manager is :ref:`configured <config_overload_manager>` by specifying a set of
 resources to monitor and a set of overload actions that will be taken when some of those
 resources exceed certain pressure thresholds.
+
+Architecture
+------------
+
+The overload manager works by periodically polling the *pressure* of a set of **resources**,
+feeding those through **triggers**, and taking **actions** based on the triggers. The set of
+resource monitors, triggers, and actions are specified at startup.
+
+Resources
+~~~~~~~~~
+
+A resource is a thing that can be monitored by the overload manager, and whose *pressure* is
+represented by a real value in the range [0, 1]. The pressure of a resource is evaluated by a
+*resource monitor*. See the :ref:`configuration page <config_overload_manager>` for setting up
+resource monitors.
+
+Triggers
+~~~~~~~~
+
+Triggers are evaluated on each resource pressure update, and convert a resource pressure value
+into one of two action states:
+
+.. _arch_overview_overload_manager-triggers-state:
+
+.. csv-table::
+  :header: state, description
+  :widths: 1, 2
+
+  saturated, the resource pressure is at or above the critical point; drastic action should be taken
+  scaling,   the resource pressure is below the critical point; action may be taken
```

Comment:
Should we also have an "inactive" state so we can distinguish when envoy is not at all in overload so no action will be taken vs "scaling" when envoy is in partial overload so some scaled action will be taken? You refer to such an "inactive" state in the OverloadAction proto below.

Reply:
I removed it from code after talking to @antoniovicente. I think the expectation is that actions that respond to active/inactive can just check for "active or not" and actions that look at the scaling value can check for "value() == 0"

Reply:
Ok, then you should probably remove the reference to the "inactive" state in the proto comments.

Reply:
One last argument for the "inactive" state - it might not be so useful in the code that implements the overload actions but it would be useful in logging and monitoring to distinguish between "inactive" and "scaling". During normal operation I think we'd expect all overload actions to be in "inactive" state and it would be useful to know when we transition out of that state.

Reply:
Yeah I keep going back and forth. It feels awkward in the documentation to say 0 is still scaling. WDYT @eziskind of doing this in a follow-on PR?

Reply:
Sure, a follow-on PR is fine with me.